### ISSUE TITLE
Name not showing in Singapore-created EUDCC

### ISSUE THREAD

**Issue by ministiy** [admin-ch/CovidCertificate-App-Android#370]
Hi!

I am based in Singapore and recently our government and EU have treated either certificates as equal. I am trying to store the Singapore-produced QR code to the app but got a **null null** as my name. All the other data seems to be the same compared to this [test data](https://github.com/admin-ch/dgc-testdata/tree/main/CH/png). The Covid Check seems to be showing my details just fine but not the Covid Wallet/Cert

# Environment
App Version 3.1.0
Android 10
EMUI 10.1.0 build 10.1.0.298

Just want to raise the issue in case it hasn't been noticed before, but realized the implementation is not that long ago.

Really great work by the way, love that the app is open sourced on Github!

**Comment by goebelUB** [comment]
Thanks for your report! Looking at [Singapore's test files](https://github.com/eu-digital-green-certificates/dgc-testdata/tree/main/SG) it seems that they only have the standardised surname set. Formally that's okay since this is the only field in the name that the [EU spec requires](https://github.com/ehn-dcc-development/ehn-dcc-schema/blob/release/1.3.0/DCC.Core.Types.schema.json#L17).

The wallet app uses the firstname (`fn`) and surname (`gn`) fields to display the name. Since those are not set, they default to null.
The verifier app always shows the standardised name next to the "simple name", but the wallet app currently does not (see #289).

Thus in addition to #289, we can probably also improve this by falling back to the standardised name in the wallet app if the "simple name" is not set.

**Comment by goebelUB** [comment]
@ministiy The fix is planned to be in the next version (3.3.0). For pretty screenshots, see [here](https://github.com/admin-ch/CovidCertificate-App-Android/pull/376). Thanks again for reporting!

**Comment by ministiy** [comment]
That looks perfect. Thanks again for the amazing support and keep up the open source spirit!

**Comment by ministiy** [comment]
Hi @goebelUB 

Sorry to comment on closed issue, I just installed the V3.3.0 from Github but still have the same issue. Trying to pinpoint the issue but to no avail. Comparing my QR code JSON data with the test data, it seems to be the same. Can provide you with details if needed. I have tried clearing cache, uninstalling and installing it again but issue still persist.

**Comment by goebelUB** [comment]
You're right, somehow the fix was merged into the main branch but not into the release/3.3.0 branch that we had already open in parallel.
It's definitely fixed on the current main branch so it should make it into the next (3.4.0) release. Sorry about that!

**Comment by goebelUB** [comment]
Can you please try again with v3.5.0? The patch should have been included there.
